collections/guides/quickstarts/quickstart_java.md

```diff
@@ -0,0 +1,234 @@
+    .header("BT-API-KEY", "key_E6gVJEn1DRT5m6YHs55Dsa")
+    .build();
+
+var getResponse = client.send(getRequest, BodyHandlers.ofString());
+System.out.println(getRequest);
+System.out.println(getResponse.body());
+```
+
+## Step 7: Run the application
+Test the entire tutorial out by running the application:
+
+```bash
+javac BasisTheoryQuickstart.java
+java BasisTheoryQuickstart
+```
+
+🎉🎉🎉 You’ve successfully created a Token for your data and read it back 🎉🎉🎉
+
+```bash
+> Create a token:
+{"id":"62a71684-f148-424c-88c6-bdb44031357a","tenant_id":"570b53fb-1ecf-4aaf-9cb2-145e13b566a9","type":"token","privacy":{"classification":"general","impact_level":"high","restriction_policy":"redact"},"created_by":"7281f0ef-eafc-455c-bdae-ce6c99ff8268","created_at":"2022-02-17T21:49:29.2596915+00:00"}
+Read a token:
+{"id":"62a71684-f148-424c-88c6-bdb44031357a","type":"token","tenant_id":"570b53fb-1ecf-4aaf-9cb2-145e13b566a9","created_by":"7281f0ef-eafc-455c-bdae-ce6c99ff8268","created_at":"2022-02-17T21:49:29.2596915+00:00","privacy":{"classification":"general","impact_level":"high","restriction_policy":"redact"}}
+```
+
+## Put it all together
+This completes the basic ability to secure data with Tokens and retrieve the raw data back from Basis Theory when you need to use the data in your systems. This flow allows you to secure your data at rest, removes the liability of having the data stored in your databases, and frees you from having to worry about complex encryption logic.
+
+```java
+import java.net.URI;
+import java.net.http.HttpClient;
+import java.net.http.HttpRequest;
+import java.net.http.HttpRequest.BodyPublishers;
+import java.net.http.HttpResponse.BodyHandlers;
+
+public class BTQuickstart {
+
+    public static void main(String[] args) {
+        try {
+            // Initialize the HttpClient
+            var client = HttpClient.newHttpClient();
+
+            // JSON message formatted as a string
+            var message = "{\"type\": \"token\", \"data\": \"foo\"}";
+
+            var createRequest = HttpRequest.newBuilder()
+                .uri(URI.create("https://api.basistheory.com/tokens"))
+                .POST(BodyPublishers.ofString(message))
+                .header("Content-type", "application/json")
+                .header("Accept", "application/json")
+                // Enter your API key here!
+                .header("BT-API-KEY", "key_E6gVJEn1DRT5m6YHs55Dsa")
+                .build();
+
+            // Create a new token
+            var createResponse = client.send(createRequest, BodyHandlers.ofString());
+            var token = createResponse.body();
+
+            // Token response
+            System.out.println("Create a token:");
+            System.out.println(token);
+
+            // Parse token id from response
+            var tokenArray = token.split("\"");
+            var tokenId = tokenArray[3];
+
+            // Get token
+            var getRequest = HttpRequest.newBuilder()
```

Comment:
The request logic could be abstracted but since this is a simple example it seems fine

Reply:
i went back and forth a few times on this, and decided to keep it extremely simple for this quickstart. unlike the other guides, Java requires the most external help, so it was going to be more complex either way.